cpp/src/arrow/compare.cc

```diff
@@ -782,6 +783,96 @@ bool TensorEquals(const Tensor& left, const Tensor& right) {
   return are_equal;
 }
 
+namespace {
+
+template <typename LeftSparseIndexType, typename RightSparseIndexType>
+struct SparseTensorEqualsImpl {
+  static bool compare(const SparseTensor<LeftSparseIndexType>& left,
```

Comment:
We use CapWords for non-trivial function names. This is OK for now because it is not public